pkg/controller/networkpolicy/validate.go

```diff
@@ -0,0 +1,238 @@
+// Copyright 2019 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package networkpolicy
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	admv1 "k8s.io/api/admission/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/klog/v2"
+
+	secv1alpha1 "github.com/vmware-tanzu/antrea/pkg/apis/security/v1alpha1"
+)
+
+type NetworkPolicyValidator struct {
+	networkPolicyController *NetworkPolicyController
+}
+
+// NewNetworkPolicyValidator returns a new *NetworkPolicyValidator.
+func NewNetworkPolicyValidator(networkPolicyController *NetworkPolicyController) *NetworkPolicyValidator {
+	return &NetworkPolicyValidator{
+		networkPolicyController: networkPolicyController,
+	}
+}
+
+func HandleValidationNetworkPolicy(v *NetworkPolicyValidator) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		klog.Info("Received request to validate Tier/CNP CRD")
+		var reqBody []byte
+		if r.Body != nil {
+			reqBody, _ = ioutil.ReadAll(r.Body)
+		}
+		if len(reqBody) == 0 {
+			klog.Errorf("Tier validation received empty request body")
+			http.Error(w, "empty request body", http.StatusBadRequest)
+			return
+		}
+		// verify the content type is accurate
+		contentType := r.Header.Get("Content-Type")
+		if contentType != "application/json" {
+			klog.Errorf("Invalid content-Type=%s, expected application/json", contentType)
+			http.Error(w, "invalid Content-Type, expected `application/json`", http.StatusUnsupportedMediaType)
+			return
+		}
+		if r.URL.Path != "/validate/tier" && r.URL.Path != "/validate/cnp" {
```

Comment:
nit: This should never happen as we only register this function with those two paths?

Reply:
There's no harm in adding this here though. Thanks for the explanation.